lib/ansible/modules/cloud/vmware/vcenter_license.py

```diff
@@ -177,8 +199,25 @@ def main():
         if key is not None:
             lam = lm.licenseAssignmentManager
             assigned_license = None
+            datacenter = module.params['datacenter']
+            datacenter_obj = None
+            if datacenter:
+                datacenter_obj = pyv.find_datacenter_by_name(datacenter)
```

Comment:
You can raise the error here:

```python
    if not datacenter_obj:
        module.fail_json(msg="Unable to find the datacenter %(datacenter)s" % module.params)
```

This way, you simplify the logic of the "cluster" block.

Reply:
OK.